Removes mention of hybrid, rewrites rendering pages (#9339)
withastro/docs: src/content/docs/en/guides/on-demand-rendering.mdx

<IntegrationsNav category="adapter"/>

### Add an Adapter

To render any of your pages on demand, for any `output` mode, you need to add an **adapter**.

Each adapter allows Astro to output a script that runs your project on a specific *runtime*: the environment that runs code on the server to generate pages when they are requested (e.g. Netlify, Cloudflare).

You can find both [official and community adapters in our integrations directory](https://astro.build/integrations/?search=&categories%5B%5D=adapters). Choose the one that corresponds to your [deployment environment](/en/guides/deploy/).

You can add any of the [official adapters maintained by Astro](/en/guides/integrations-guide/#official-integrations) with the following `astro add` command. This will install the adapter and make the appropriate changes to your `astro.config.mjs` file in one step. 

For example, to install the Netlify adapter, run:

<PackageManagerTabs>
  <Fragment slot="npm">
  ```shell
  npx astro add netlify
  ```
  </Fragment>
  <Fragment slot="pnpm">
  ```shell
  pnpm astro add netlify
  ```
  </Fragment>
  <Fragment slot="yarn">
  ```shell

>   yarn astro add net

Reviewer: I think the name is the same for yarn:
```suggestion
  yarn astro add netlify
```

Author: hahaha, I actually stopped to tell someone I was updating these to Netlify, then forgot to come back and finish

---
title: On-demand Rendering
i18nReady: true
---
import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';
import { Steps } from '@astrojs/starlight/components';
import RecipeLinks from '~/components/RecipeLinks.astro';
import IntegrationsNav from '~/components/IntegrationsNav.astro';

Astro allows you to choose **on-demand rendering** for some, or all of your pages and endpoints. 

Generating HTML pages on the server when requested and sending them to the client is also known as **server-side rendering (SSR)**. An **adapter** is used to run your project on the server and handle these requests.

## Official Adapters

Astro maintains official adapters for [Node.js](https://nodejs.org/), [Vercel](https://vercel.com/), [Netlify](https://www.netlify.com/), and [Cloudflare](https://www.cloudflare.com/). 

Find even more [community-maintained adapters](https://astro.build/integrations/?search=&categories%5B%5D=adapters) (e.g. Deno, SST, AWS) in our integrations directory. 

<IntegrationsNav category="adapter"/>

### Add an Adapter

To render any of your pages on demand, for any `output` mode, you need to add an **adapter**.

> Each adapter allows Astro to output a script that runs your project on a specific *runtime* (e.g. Netlify, Cloudflare): the environment that runs code on the server to generate pages when they are requested.

Author: ```suggestion
Each adapter allows Astro to output a script that runs your project on a specific *runtime*: the environment that runs code on the server to generate pages when they are requested (e.g. Netlify, Cloudflare).
```

You can find both [official and community adapters in our integrations directory](https://astro.build/integrations/?search=&categories%5B%5D=adapters). Choose the one that corresponds to your [deployment environment](/en/guides/deploy/).

#### `astro add` install

You can add any of the [official adapters maintained by Astro](/en/guides/integrations-guide/#official-integrations) with the following `astro add` command. This will install the adapter and make the appropriate changes to your `astro.config.mjs` file in one step. 

For example, to install the Netlify adapter, run:

> 
> <PackageManagerTabs>
>   <Fragment slot="npm">
>   ```shell
>   npx astro add netlify
>   ```
>   </Fragment>
>   <Fragment slot="pnpm">
>   ```shell
>   pnpm astro add netlify
>   ```
>   </Fragment>
>   <Fragment slot="yarn">
>   ```shell
>   yarn astro add net
>   ```
>   </Fragment>
> </PackageManagerTabs>
> 
> #### Manual Install
> 
> You can also add an adapter manually by installing the package and updating `astro.config.mjs` yourself.
> 
> For example, to install the Netlify adapter manually:
> 
> <Steps>
> 1. Install the adapter to your project dependencies using your preferred package manager:
> 
>    <PackageManagerTabs>
>      <Fragment slot="npm">
>      ```shell
>      npm install @astrojs/netlify
>      ```
>      </Fragment>
>      <Fragment slot="pnpm">
>      ```shell
>      pnpm add @astrojs/netlify
>      ```
>      </Fragment>
>      <Fragment slot="yarn">
>      ```shell
>      yarn add @astrojs/netlify
>      ```
>      </Fragment>
>    </PackageManagerTabs>
> 
> 2. [Add the adapter](/en/reference/configuration-reference/#adapter) to your `astro.config.mjs` file's import and default export:
> 
>     ```js title="astro.config.mjs" ins={2,5}
>     import { defineConfig } from 'astro/config';
>     import netlify from '@astrojs/netlify';
> 
>     export default defineConfig({
>       adapter: netlify(),
>     });
>     ```
> 
>     Note that different adapters may also have different configuration settings. Read each adapter's documentation, and apply any necessary config options to your chosen adapter in `astro.config.mjs`
> </Steps>
> 

Author: ```suggestion
#### `astro add` install

You can add any of the [official adapters maintained by Astro](/en/guides/integrations-guide/#official-integrations) with the following `astro add` command. This will install the adapter and make the appropriate changes to your `astro.config.mjs` file in one step. 

For example, to install the Netlify adapter, run:

<PackageManagerTabs>
  <Fragment slot="npm">
  ```shell
  npx astro add netlify
  ```
  </Fragment>
  <Fragment slot="pnpm">
  ```shell
  pnpm astro add netlify
  ```
  </Fragment>
  <Fragment slot="yarn">
  ```shell
  yarn astro add net
  ```
  </Fragment>
</PackageManagerTabs>

You can also [add an adapter manually by installing the NPM package](/en/guides/integrations-guide/#installing-an-npm-package) (e.g. `@astrojs/netlify`) and updating `astro.config.mjs` yourself.

Note that different adapters may have different configuration settings. Read each adapter's documentation, and apply any necessary config options to your chosen adapter in `astro.config.mjs`
```
Don't need to show the example and take up room on this page, when we have an example of installing an integration manually! Remove lines!

For example, to install the Netlify adapter, run:

<PackageManagerTabs>
  <Fragment slot="npm">
  ```shell
  npx astro add netlify
  ```
  </Fragment>
  <Fragment slot="pnpm">
  ```shell
  pnpm astro add netlify
  ```
  </Fragment>
  <Fragment slot="yarn">
  ```shell
  yarn astro add net
  ```
  </Fragment>
</PackageManagerTabs>

#### Manual Install

You can also add an adapter manually by installing the package and updating `astro.config.mjs` yourself.

For example, to install the Netlify adapter manually:

<Steps>
1. Install the adapter to your project dependencies using your preferred package manager:

   <PackageManagerTabs>
     <Fragment slot="npm">
     ```shell
     npm install @astrojs/netlify
     ```
     </Fragment>
     <Fragment slot="pnpm">
     ```shell
     pnpm add @astrojs/netlify
     ```
     </Fragment>
     <Fragment slot="yarn">
     ```shell
     yarn add @astrojs/netlify
     ```
     </Fragment>
   </PackageManagerTabs>

> 2. [Add the adapter](/en/reference/configuration-reference/#adapter) to your `astro.config.mjs` file's import and default export, along with your desired `output` mode:

Author: ```suggestion
2. [Add the adapter](/en/reference/configuration-reference/#adapter) to your `astro.config.mjs` file's import and default export:
```

>     ```js ins={3,7} {6}
>     // astro.config.mjs
>     import { defineConfig } from 'astro/config';
>     import netlify from '@astrojs/netlify';
> 
>     export default defineConfig({
>       output: 'server',
>       adapter: netlify(),
>     });
>     ```

Author: ``````suggestion
    ```js title="astro.config.mjs" ins={2,5}
    import { defineConfig } from 'astro/config';
    import netlify from '@astrojs/netlify';

    export default defineConfig({
      adapter: netlify(),
    });
    ```
``````
